telegram/request/_requestparameter.py

```diff
@@ -115,6 +115,11 @@ def _value_and_input_files_from_input(  # pylint: disable=too-many-return-statem
         """
         if isinstance(value, dtm.datetime):
             return to_timestamp(value), []
+        if isinstance(value, dtm.timedelta):
+            seconds = value.total_seconds()
+            if seconds.is_integer():
+                return int(seconds), []
+            return value.total_seconds(), []
```

Comment:
this is a little confusing. total_seconds always returns an integer right?

Reply:
not necessarily, as the delta could also have microseconds

Reply:
I see. But then the logic should be reversed?

```suggestion
            seconds = value.total_seconds()  # could return seconds with microseconds (a float)
            if seconds.is_integer():  # if it's only seconds
                return seconds, []
            return int(seconds), []  # if it's a float, round down to the nearest integer
```

Reply:
No, the `int` is mainly there for mypy. I don't want to cut off the granularity. We probably don't use it anywhere right now, but this feels like the thing that will be a nightmare to debug if we ever need it and then have to track down where there decimal points are lost 😬

Reply:
hmm, but does the Telegram API accept it if `float`s are passed in the field? The `WithRequest*` tests only tested with seconds and not milliseconds iisc.

Reply:
I'm not sure tbh. So far I would have expected them to not accept. I can double check that. My pov is rather that we should not make any silent conversions. If the user passes nonsense, they should get an understandable error message. we also don't validate any other input

Reply:
fair enough